### PR TITLE
[Spark-700] Example making Spark RDD from Cassandra Column Family

### PR DESCRIPTION
**Pull request by milliondreams**
Sorry missed mentioning earlier. This will close https://spark-project.atlassian.net/browse/SPARK-700

### CODE REVIEW

**Comment by milliondreams** [comment]
Updated the example to demonstrate RDD save to Cassandra

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -201,8 +201,8 @@ object SparkBuild extends Build {
 
   def examplesSettings = sharedSettings ++ Seq(
     name := "spark-examples",
-    libraryDependencies ++= Seq("com.twitter" % "algebird-core_2.9.2" % "0.1.11")
-  )
+    libraryDependencies ++= Seq("com.twitter" % "algebird-core_2.9.2" % "0.1.11",
+      "org.apache.cassandra" % "cassandra-all" % "1.2.5" exclude("com.google.guava", "guava") exclude("com.googlecode.concurrentlinkedhashmap", "concurrentlinkedhashmap-lru") exclude("com.ning","compress-lzf") exclude("io.netty","netty") exclude("jline","jline") exclude("log4j","log4j") exclude("org.apache.cassandra.deps", "avro")))
```
Comment (pwendell) [inline project/SparkBuild.scala:205]: Is there any cassandra maven package that has only client libraries?

File: examples/src/main/scala/spark/examples/CassandraTest.scala [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import java.nio.ByteBuffer
+import java.util.SortedMap
+import org.apache.cassandra.db.IColumn
+import org.apache.cassandra.utils.ByteBufferUtil
+import scala.collection.JavaConversions._
+
+
+
+/*
+ * This example demonstrates using Spark with Cassandra with the New Hadoop API and Cassandra support for Hadoop.
```
Comment (pwendell) [inline examples/src/main/scala/spark/examples/CassandraTest.scala]: Is this > 100 characters? If so, please wrap it or shorten the line. Same goes for all other lines...

File: examples/src/main/scala/spark/examples/CassandraTest.scala [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    sliceRange.setStart(Array.empty[Byte])
+    sliceRange.setFinish(Array.empty[Byte])
+    predicate.setSlice_range(sliceRange)
+    ConfigHelper.setInputSlicePredicate(job.getConfiguration(), predicate)
+
+    ConfigHelper.setInputPartitioner(job.getConfiguration(), "Murmur3Partitioner")
+
+    ConfigHelper.setOutputPartitioner(job.getConfiguration(), "Murmur3Partitioner")
+
+    //Make a new Hadoop RDD
```
Comment (pwendell) [inline examples/src/main/scala/spark/examples/CassandraTest.scala]: Space after //

File: examples/src/main/scala/spark/examples/CassandraTest.scala [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      classOf[SortedMap[ByteBuffer, IColumn]])
+
+    // Let us first get all the paragraphs from the retrieved rows
+    val paraRdd = casRdd.map {
+      case (key, value) => {
+        ByteBufferUtil.string(value.get(ByteBufferUtil.bytes("para")).value())
+      }
+    }
+
+    //Lets get the word count in paras
```
Comment (pwendell) [inline examples/src/main/scala/spark/examples/CassandraTest.scala]: Space after //

**Comment by pwendell** [comment]
This will be useful to have, thanks for adding this! A few notes about style inline. I also noticed that there are some double-blank-lines in various places that don't seem to separate major sections. It might be good to clean those up to make this more consistent with our style guide.

File: project/SparkBuild.scala (hunk 1 above)
Comment (milliondreams) [inline project/SparkBuild.scala:205]: cassandra-thrift has just the cassandra client libraries but sadly they don't include the Hadoop support files in it.

**Comment by milliondreams** [comment]
Fixed the style.

**Comment by mateiz** [comment]
Thanks Rohit -- I've merged this in.

**Comment by mohammedguller** [comment]
Rohit/Matei - it would be great if you could update the example to use the new second generation of Cassandra Hadoop driver, which uses new classes such as CqlPagingInputFormat, CqlPagingRecordReader, CqlOutputFormat and CqlRecordWriter. 

Thanks.

**Comment by LannyRipple** [comment]
@mohammedguller Many older systems are still using the Thrift interface and can't move away because the data cannot be migrated to a CQL based C\* installation.

If a Cql based example is created please also keep the Thrift based one.

**Pull request merged**